ESP_ATMod/command.cpp

```diff
@@ -1643,6 +1669,165 @@ void cmd_AT_CIPMUX()
 	}
 }
 
+/*
+ * AT+CIPCIPSERVER - Deletes/Creates TCP Server
+ */
+void cmd_AT_CIPSERVER()
+{
+
+	if (!gsCipMux)
+	{
+		Serial.printf_P(MSG_ERROR);
+		return;
+	}
+	uint8_t error = 1; // 1 = generic error, 0 = ok
+	uint16_t offset = strlen("AT+CIPSERVER");
+
+	bool stop = false;
+	uint32_t port = 0;
+
+	do
+	{
+		if (inputBuffer[offset] != '=')
+			break;
+
+		++offset;
+		stop = inputBuffer[offset] == '0';
+		++offset;
+
+		if (inputBufferCnt > offset)
+		{
+			if (inputBuffer[offset] != ',')
+				break;
+
+			++offset;
+			error = 2;
+
+			if (!readNumber(inputBuffer, offset, port) || port > 65535)
+				break;

```

Comment:
There is no end of buffer checking, so I can type `AT+CIPSERVER=1,8080,ThisIsAnError` and it works. Should be ERROR,, no extra parameters allowed.

Reply:
sorry but I don't see such check in other commands processing. I did it by example, not my way.

Reply:
ok I see that the check is `&& inputBufferCnt == offset + 2`